scripts/hostcfgd

```diff
@@ -284,8 +294,47 @@ class FeatureHandler(object):
             self.disable_feature(feature)
             syslog.syslog(syslog.LOG_INFO, "Feature {} is stopped and disabled".format(feature.name))
 
+        self.sync_feature_asic_scope(feature)
+
         return True
 
+    def sync_feature_asic_scope(self, feature_config):
+        """Updates the has_per_asic_scope field in the FEATURE|* tables as the field
+        might have to be rendered based on DEVICE_METADATA table or Device Running configuration.
+        Disable the ASIC instance service unit file it the render value is False and update config 
+
+        Args:
+            feature: An object represents a feature's configuration in `FEATURE`
+            table of `CONFIG_DB`.
+
+        Returns:
+            None.
+        """
+
+        cmds = []
+        feature_names, feature_suffixes = self.get_multiasic_feature_instances(feature_config, True)
+        for feature_name in feature_names:
+            if '@' not in feature_name:
+                continue
+            unit_file_state = self.get_systemd_unit_state("{}.{}".format(feature_name, feature_suffixes[-1]))
+            if not unit_file_state:
+                continue
+            if unit_file_state != "disabled" and not feature_config.has_per_asic_scope:
+                for suffix in reversed(feature_suffixes):
+                    cmds.append("sudo systemctl stop {}.{}".format(feature_name, suffix))
+                    cmds.append("sudo systemctl disable {}.{}".format(feature_name, feature_suffixes[-1]))
+                    cmds.append("sudo systemctl mask {}.{}".format(feature_name, feature_suffixes[-1]))
+                for cmd in cmds:
+                    syslog.syslog(syslog.LOG_INFO, "Running cmd: '{}'".format(cmd))
+                    try:
+                        run_cmd(cmd, raise_exception=True)
+                    except Exception as err:
+                        syslog.syslog(syslog.LOG_ERR, "Feature '{}.{}' failed to be stopped and disabled"
+                                      .format(feature.name, feature_suffixes[-1]))
+                        self.set_feature_state(feature, self.FEATURE_STATE_FAILED)
+                        return
+        self._config_db.mod_entry('FEATURE', feature_config.name, {'has_per_asic_scope': str(feature_config.has_per_asic_scope)})
```

Comment:
The host config_db will have `has_per_asic_scope` as failed but the config_db for the asic will have them as enabled? should we update that as well to avoid confusion

Reply:
@arlakshm yes, that we can do later if needed. Currently also other fields of this table are not same between host and asic.
I think that change will be coming as part of separate PR may be where we needed to update (ex:macsec feature state) that need be updated in asic db also. In that PR we can update this field also.

@judyjoseph